app/assets/javascripts/devices/devices-show.es6

```diff
@@ -0,0 +1,182 @@
+          var url = $editable.data('url');
+          var data = { checkin: { lat: coords[0], lng: coords[1]} }
+          putUpdateCheckin(url, data);
+        } else {
+          // reverse the edit
+          $editable.text(original);
+        }
+        handleEditEnd($editable);
+      }
+
+      function handleMapClick($editable, e) {
+        var confirmText = "Are you sure? Click ok to reposition check-in to new coordinates (";
+            confirmText += e.latlng.lat.toFixed(6) + ", " + e.latlng.lng.toFixed(6) + ").";
+        if (confirm(confirmText)) {
+          var data = { checkin: {lat: e.latlng.lat, lng: e.latlng.lng} }
+          putUpdateCheckin($editable.data('url'), data);
+        }
+        handleEditEnd($editable);
+      }
+
+      function putUpdateCheckin(url, data) {
+        $.ajax({
+          dataType: 'json',
+          url: url,
+          type: 'PUT',
+          data: data
+        })
+        .done(updateCheckin)
+        .fail(function (error) {
+          console.log('Error updating checkin:', error);
+        })
+      }
+
+      function updateCheckin(response) {
+        // tries to find the checkin in gon and update it with the response
+        checkin = _.find(gon.checkins, _.matchesProperty('id',response.id));
+        checkin.lat = response.lat;
+        checkin.lng = response.lng;
+        checkin.edited = response.edited;
+        checkin.lastEdited = true;
+        checkin.address = response.address;
+        checkin.fogged_city = response.fogged_city;
+        // delete the localDate so we generate fresh timezone data
+        delete checkin.localDate;
+        M.refreshMarkers(gon.checkins);
+      }
+
+      function handleEditEnd($editable) {
+        map.removeControl(COPO.maps.mousePositionControl);
+        $('#map').toggleClass('crosshair');
+        $editable.removeAttr('contenteditable');
+        U.deselect();
+        $('.editable-wrapper').toggleClass('clickable');
+        unsetEditableListeners($editable)
+      }
+
+      function unsetEditableListeners($editable) {
+        map.off('click');
+        $('.leaflet-popup').off('click');
+        $editable.off('keydown');
+      }
+
+    }
+
+    function postLocation(position) {
+      $.ajax({
+        url: '/users/'+gon.current_user_id+'/devices/'+gon.device+'/checkins/',
+        type: 'POST',
+        dataType: 'script',
+        data: { checkin: { lat: position.coords.latitude, lng: position.coords.longitude, fogged: fogged } }
+      });
+    }
+
+    function getLocation(fogged) {
+      if(currentCoords) {
+        var position = { coords: { latitude: currentCoords.lat, longitude: currentCoords.lng } }
+        postLocation(position)
+      } else {
+        navigator.geolocation.getCurrentPosition(postLocation, U.geoLocationError, { timeout: 5000 });
+      }
+    }
+
+    function onLocationFound(p) {
+      currentCoords = p.latlng;
+    }
+  }
+});
```

Comment:
Also, if no lack of time and since this file is quite big, I'd suggest to think about how to split it into several modules (with extract entity or some other technics).